### PR TITLE
Log only span context

### PR DESCRIPTION
**Pull request by yurishkuro**
Resolves #123

### CODE REVIEW

**Comment by yurishkuro** [comment]
@tiffon @saminzadeh could you please take a look at the build failures? All seem to be related to the error below, a transitive dependency of `request` module. There was a recent change there https://github.com/request/request/pull/2751. How can we fix it if we're using babel?

```
> mocha --compilers js:babel-register crossdock/test
/home/travis/build/uber/jaeger-client-node/node_modules/request/node_modules/hawk/node_modules/boom/lib/index.js:5
const Hoek = require('hoek');
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Module._extensions..js (module.js:474:10)
    at Object.require.extensions.(anonymous function) [as .js] (/home/travis/build/uber/jaeger-client-node/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
```

**Comment by tiffon** [comment]
@yurishkuro Looks related to running node `0.10.x` which doesn't support `const` without a flag. [Details.](https://stackoverflow.com/questions/22603078/syntaxerror-use-of-const-in-strict-mode)

I would say we can either file a bug with request because their upgrade is a breaking change or change our `package.json` to get rid of the `^` in `"request": "^2.74.0"`. Generally, I would lean toward the latter or both.

File: src/reporters/logging_reporter.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default class LoggingReporter {
     }
 
     report(span: Span): void {
-        this._logger.info(`Reporting span ${JSON.stringify(span)}`);
+        this._logger.info(`Reporting span ${span.context().toString()}`);
```
Comment (tiffon) [inline src/reporters/logging_reporter.js]: Interpolated values are automatically converted to strings:

```javascript
`${{toString: () => 'abc'}}`
// -> "abc"
```

Comment (yurishkuro) [reply src/reporters/logging_reporter.js]: I don't follow. IIRC Javascript does not have Java's convention of automatically calling `toString()` method when converting to strings.

**Comment by yurishkuro** [comment]
> Looks related to running node `0.10.x` which doesn't support `const` without a flag.

The `request` module declares `"node": ">= 4"` in its package.json, so they are free to say they do not intend to support 0.10.

It seems like we're only using `request` in the crossdock tests, couldn't we use something else that is 0.10 compatible?

File: src/reporters/logging_reporter.js (hunk 1 above)
Comment (tiffon) [reply src/reporters/logging_reporter.js]: JS does call `toString()` when converting to string.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString

>Every object has a toString() method that is automatically called when the object is to be represented as a text value or when an object is referred to in a manner in which a string is expected.

**Comment by saminzadeh** [comment]
@yurishkuro Can't we just build the src code with Node 4 (or above) and pipe it through babel so its compatible with Node 0.10?

**Comment by yurishkuro** [comment]
@saminzadeh we can, but it makes the build a lot more complicated because we want the unit and integration tests run on the actual Node version from Travis matrix. Besides, this `request` module is a runtime dependency of the crossdock tests, so I don't believe what you propose would've helped with the `const` issue.

**Pull request merged**